test/hotspot/jtreg/serviceability/jvmti/events/ClassPrepare/classprep01/libclassprep01.cpp

```diff
@@ -0,0 +1,380 @@
+  int i;
+
+  err = jvmti->GetClassSignature(cls, &inf.sig, &generic);
+  if (err != JVMTI_ERROR_NONE) {
+    LOG("(GetClassSignature#%" PRIuPTR ") unexpected error: %s (%d)\n",
+           eventsCount, TranslateError(err), err);
+    result = STATUS_FAILED;
+    return;
+  }
+
+  // filter only events for test classes using classprep01 as a prefix
+  // there are a lot of classes might be generated and loaded
+  if (strncmp("Lclassprep01", inf.sig, 12) !=0) {
+    return;
+  }
+  err = jvmti->GetClassStatus(cls, &inf.status);
+  if (err != JVMTI_ERROR_NONE) {
+    LOG("(GetClassStatus#%" PRIuPTR ") unexpected error: %s (%d)\n",
+           eventsCount, TranslateError(err), err);
+    result = STATUS_FAILED;
+  }
+  err = jvmti->GetClassMethods(cls, &inf.mcount, &methods);
+  if (err != JVMTI_ERROR_NONE) {
+    LOG("(GetClassMethods#%" PRIuPTR ") unexpected error: %s (%d)\n",
+           eventsCount, TranslateError(err), err);
+    result = STATUS_FAILED;
+    return;
+  }
+  err = jvmti->GetClassFields(cls, &inf.fcount, &fields);
+  if (err != JVMTI_ERROR_NONE) {
+    LOG("(GetClassMethods#%" PRIuPTR ") unexpected error: %s (%d)\n",
+           eventsCount, TranslateError(err), err);
+    result = STATUS_FAILED;
+    return;
+  }
+  err = jvmti->GetImplementedInterfaces(cls, &inf.icount, &interfaces);
+  if (err != JVMTI_ERROR_NONE) {
+    LOG("(GetImplementedInterfaces#%" PRIuPTR ") unexpected error: %s (%d)\n",
+           eventsCount, TranslateError(err), err);
+    result = STATUS_FAILED;
+    return;
+  }
+
+  LOG(">>> [class prepare event #%" PRIuPTR "]", eventsCount);
+  LOG(" \"%s\"\n", inf.sig);
+  LOG(">>> Got ClassPrep event in thread.\n");
+  print_thread_info(jvmti, jni, thr);
+  LOG(">>>   status: ");
+  printStatus(inf.status);
+  LOG(">>>   %d methods:", inf.mcount);
+  for (i = 0; i < inf.mcount; i++) {
+    if (i > 0) LOG(",");
+    if (methods[i] == NULL) {
+      LOG(" null");
+    } else {
+      err = jvmti->GetMethodName(methods[i], &name, &sig, &generic);
+      if (err == JVMTI_ERROR_NONE) {
+        LOG(" \"%s%s\"", name, sig);
+      } else {
+        LOG(" ???");
+      }
+    }
+  }
+  LOG("\n");
+  LOG(">>>   %d fields:", inf.fcount);
+  for (i = 0; i < inf.fcount; i++) {
+    if (i > 0) LOG(",");
+    if (fields[i] == NULL) {
+      LOG(" null");
+    } else {
+      err = jvmti->GetFieldName(cls, fields[i],
+                                    &name, &sig, &generic);
+      if (err == JVMTI_ERROR_NONE) {
+        LOG(" \"%s, %s\"", name, sig);
+      } else {
+        LOG(" ???");
+      }
+    }
+  }
+  LOG("\n");
+  LOG(">>>   %d interfaces:", inf.icount);
+  for (i = 0; i < inf.icount; i++) {
+    if (i > 0) LOG(",");
+    if (interfaces[i] == NULL) {
+      LOG(" null");
+    } else {
+      err = jvmti->GetClassSignature(interfaces[i], &sig, &generic);
+      if (err == JVMTI_ERROR_NONE) {
+        LOG(" \"%s\"", sig);
+      } else {
+        LOG(" ???");
+      }
+    }
+  }
+  LOG("\n");
+
+
```

Comment:
Too many empty lines. It might make sense to do a common cleanup with all edits like this.